### DISCUSSION TITLE
Is the use of Docker always necessary when using GitHub Packages?

### DISCUSSION THREAD

**Discussion by Rod-at-DOH** [skills/.github#222] [Q&A] [answered]
I was going through the **GitHub Skill on GitHub Packages**. I'm sure that using packages is something we need to adopt. I got to _Step 4: Pull Your Image_, which requires installing Docker onto one's dev box. I tried doing that, but it failed miserably. I've tried (unsuccessfully) ever since to get Docker working on my dev box, which has pulled me away from finishing the GH Packages course.

What I'd like to know is this, is Docker on a dev box necessary in all cases and environments? At this point what I need to do is get some .NET Framework DLLs into a GH Package. Do I need to use Docker for that? If not, then how do I do that?

**Comment by sinsukehlab** [comment]
The Skills course "Publish to GitHub Packages" builds a Docker image, so you need a Docker environment to pull the image. You can use [Docker Playground](https://labs.play-with-docker.com/) if you don't have a local Docker environment.
If you would like to use GitHub Packages in your repos (not in the Skills course), you can also use other package managers supported by GitHub Packages. GitHub Packages support npm, RubyGems, Apache Maven, Gradle, Docker, and NuGet. NuGet would match .NET Framework DLLs.

**Reply by Rod-at-DOH** [reply]
So, @sinsukehlab, is the only way to use GitHub Packages, is to use Docker? Regardless of the "Publish to GitHub Packages" course?

**Reply by sinsukehlab** [reply, marked as answer]
No, GitHub Packages also supports other package managers. You can use NuGet if you like the .NET CLI.
https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages

**Reply by sinsukehlab** [reply]
I'm glad that I could help you out.
